### Q1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
cutlet_data = pd.read_csv("C:\\My Computer\\Documents\\Excler\\Assignment\\Hypothesis Testing\\Cutlets.csv")

In [3]:
cutlet_data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
cutlet_data.tail()

Unit A  Unit B
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

##### Using two sample two tail test
##### H0: There is no difference in diameter of the cutlet between two units
##### H1: There is difference in diameter of the cutlet between two units

In [8]:
ColA = pd.Series(cutlet_data.iloc[:,0])

In [9]:
ColB = pd.Series(cutlet_data.iloc[:,1])

In [10]:
stats.ttest_ind(ColA,ColB) #two sample two tail test for this hypothesis test

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

#### As the pvalue is greater than the suggested significance level of 5% (0.4722>0.05), we accept null hypothesis. Therefore, there is no difference in diaemter of the cutlet between two units

### Q2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
### Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

In [11]:
lab_data = pd.read_csv("C:\\My Computer\\Documents\\Excler\\Assignment\\Hypothesis Testing\\LabTAT.csv")

In [12]:
lab_data.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [13]:
lab_data.tail()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

##### Using ANOVA to test our hypothesis
##### H0: There is no difference in TAT average among the different labs
##### H1: Atleast 1 TAT average is different than the other three labs

In [14]:
stats.f_oneway(lab_data.iloc[:,0], lab_data.iloc[:,1], lab_data.iloc[:,2], lab_data.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### As the pvalue is very low compared to the given 5% significance level, we will reject null hypothesis in favor of alternative hypothesis. Therefore, there is difference in average TAT among the different laboratories

### Q3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [15]:
buyer_data = pd.read_csv("C:\\My Computer\\Documents\\Excler\\Assignment\\Hypothesis Testing\\BuyerRatio.csv")

In [16]:
buyer_data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [17]:
from scipy.stats import chi2_contingency

In [18]:
observed_values = [[50, 142, 131, 70], [435, 1523, 1356, 750]]

##### Chi-Square test
##### H0: All proportions are equal
##### H1: Not all proportions are equal

In [19]:
chi2_contingency(observed_values)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

#### As the pvalue is greater than the alpha value of 0.05, we accept null hypothesis. Therefore, all proportions are equal.

### Q4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

In [21]:
telecall = pd.read_csv("C:\\My Computer\\Documents\\Excler\\Assignment\\Hypothesis Testing\\Costomer+OrderForm.csv")

In [22]:
telecall.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [23]:
telecall.tail()

Phillippines   Indonesia       Malta       India
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

In [24]:
telecall.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [25]:
telecall.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [26]:
telecall.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [27]:
telecall.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [30]:
observed_values_telecall = [[271, 267, 269, 280],[29,33,31,20]]

##### Chi-Square test
##### H0: Defective % doesn't vary by centres
##### H1: Defective % varies by centres

In [31]:
chi2_contingency(observed_values_telecall)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

#### As the pvalue is greater than the alpha value of 0.05, we accept null hypothesis. Therefore, Defective % doesn't vary by centres.